include/fastdds/dds/builtin/typelookup/TypeLookupReplyListener.hpp

```diff
@@ -81,9 +83,39 @@ class TypeLookupReplyListener : public fastrtps::rtps::ReaderListener
     fastrtps::types::TypeObjectFactory* factory_;
 };
 
+class TypeLookupReplyWListener : public fastrtps::rtps::WriterListener
```

Comment:
Instead of creating a new pair of classes, I'd make the already existing ones to also implement the `WriterListener` interface. That way we'd have less friends, less members, less news/deletes, etc. so easier maintainability. Applies to the request listener as well